### PR TITLE
chore(mme): Split spgw_procedures_test

### LINKED ISSUE #12185: Split //lte/gateway/c/core/oai/test/spgw_task:spgw_procedures_test into multiple targets
See https://github.com/magma/magma/issues/12181 for more information on the effort

## Goals
1. Work with @magma/approvers-agw-mme (Slack: https://magmacore.slack.com/archives/C02GMJ8S8FR) to figure out how the tests should be divided.
2. For each new target, run some checks to see if the test needs to be tagged manual or flaky.

If `bazel test <TARGET> --config=asan` does not pass, file an issue to track the error and add `tags = ["manual"]`. This will ensure the test is skipped unless specified directly. 
If running `bazel test <TARGET> --runs_per_test=100`  or `bazel test <TARGET> --config=asan --runs_per_test=100` shows the target is flaky, add `flaky = True,` and file and issue to track the flakiness.

### PR DESCRIPTION
**Pull request by themarwhal**
Signed-off-by: GitHub <noreply@github.com>

## Summary
A first attempt to resolve https://github.com/magma/magma/issues/12185

This PR does the following
* Adds spgw_procedures_test_fixture.cpp/spgw_procedures_test_fixture.h to pull out test fixutre to a separate file to be shared

* Creates test_spgw_procedures_dedicated_bearer.cpp to handle all dedicated bearer related tests from test_spgw_procedures.cpp

* Create test_spgw_procedures_session.cpp to handle all session related tests from test_spgw_procedures.cpp

* Adds the test size field for all existing tests in this file  (they're all small)

*  Changes some of the test dependency from gtest_main to gtest where appropriate

## Test Plan
I ran the following commands to verify reliability
```bash
bazel test //lte/gateway/c/core/oai/test/spgw_task/... --runs_per_test=100
bazel test //lte/gateway/c/core/oai/test/spgw_task/... --runs_per_test=100 --config=asan
bazel test //lte/gateway/c/core/oai/test/spgw_task/... --runs_per_test=100 --config=lsan
```
I observed no flakiness.

I also ran `make test_oai` to make sure CMake has not regressed. (And CI)

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/core/oai/test/spgw_task/BUILD.bazel [hunk 1]
```diff
@@ -67,7 +112,7 @@ cc_test(
         ":spgw_test_core",
         "//lte/gateway/c/core",
         "//lte/gateway/c/core/oai/test/mock_tasks",
-        "@com_google_googletest//:gtest_main",
+        "@com_google_googletest//:gtest",
```
Comment (electronjoe) [inline lte/gateway/c/core/oai/test/spgw_task/BUILD.bazel:115]: I would have thought this required `gtest_main` as `test_spgw_procedures_with_injected_state.cpp` does not define a main itself. This test works with just `:gtest`?

**Review by electronjoe — APPROVED** [review]
Bazel LGTM

File: lte/gateway/c/core/oai/test/spgw_task/BUILD.bazel (hunk 1 above)
Comment (themarwhal) [reply lte/gateway/c/core/oai/test/spgw_task/BUILD.bazel:115]: Yeah I was confused about this but gtest seems to work. It did not work, however, for `//lte/gateway/c/core/oai/test/spgw_task:spgw_service_impl_test`

File: lte/gateway/c/core/oai/test/spgw_task/spgw_test_util.h [hunk 2]
```diff
@@ -23,6 +26,8 @@ extern "C" {
 namespace magma {
 namespace lte {
 
+extern task_zmq_ctx_t task_zmq_ctx_main_spgw;
```
Comment (themarwhal) [inline lte/gateway/c/core/oai/test/spgw_task/spgw_test_util.h:29]: From what I understand extern declaration should be done in header files not source.

**Review by ssanadhya — APPROVED** [review]
lgtm

**Pull request merged**